main.go

```diff
@@ -0,0 +1,62 @@
+/**
+ *Author: Stefan
+ *Date: 12/10/2019
+ *Last changes: 12/11/2019 01.19
+ *Task: Web server handling any request to it’s address and returning JSON like:
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+//Response struct
+type Response struct {
+	Host       string   `json:"host"`
+	UserAgent  string   `json:"user_agent"`
+	RequestURI string   `json:"request_uri"`
+	Headers    struct { //is correctly accoding the tasks
+		Accept    string
+		UserAgent string
+	} `json:"headers"`
+}
+
+func jsonHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	p := Response{
+		Host:       r.Host,
+		UserAgent:  r.UserAgent(),
+		RequestURI: r.RequestURI,
+		Headers: struct {
+			Accept    string
+			UserAgent string
+		}{
+			r.Header.Get("Accept"),
+			r.Header.Get("User-agent"),
+		},
+	}
+
+	//pack for Write(interface)
+	pp, _ := json.Marshal(&p)
```

Comment:
Please do not ignore an error here.
In case of error you should return it to the client and set the status code to an error.

Reply:
Done:) Thanks!

Reply:
I can return it using panic and log. What you prefer in this case?